lib/phoenix/endpoint/instrument.ex

```diff
@@ -138,7 +140,11 @@ defmodule Phoenix.Endpoint.Instrument do
     Enum.map Enum.with_index(instrumenters), fn {inst, index} ->
       quote do
         unquote(build_result_variable(index)) =
-          unquote(inst).unquote(event)(:start, var!(compile), var!(runtime))
+          try do
+            unquote(inst).unquote(event)(:start, var!(compile), var!(runtime))
+          catch
+            kind, error -> Logger.error Exception.format(kind, error)
```

Comment:
Let's add a bit of meta text here: `"Instrumenter #{instrumenter}.#{event}/3 failed.\n" <> Exception.format`. Keep in mind the string part can be generated at compile time.